### PR TITLE
Add description on failure status

### PR DESCRIPTION
**Pull request by pitming**
**What this PR does / why we need it**:
It's just add the description when the health check fails

**Which issue(s) this PR fixes**:
No issues
Please reference the issue this PR will close: #_[issue number]_

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
I don't think so

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: test/HealthChecks.Npgsql.Tests/Functional/TestLoggerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Logging;
+
+namespace HealthChecks.Npgsql.Tests.Functional;
```
Comment (sungam3r) [inline test/HealthChecks.Npgsql.Tests/Functional/TestLoggerProvider.cs:4]:
```suggestion
namespace HealthChecks.Npgsql.Tests.Functional;

```

File: test/HealthChecks.Npgsql.Tests/Functional/NpgsqlHealthCheckTests.cs [hunk 2]
```diff
@@ -152,5 +154,41 @@ public async Task be_unhealthy_if_npgsql_is_not_available_registered()
...
+                        Predicate = r => r.Tags.Contains("npgsql")
+                    });
+                });
+
+            using var server = new TestServer(webHostBuilder);
+
+            var response = await server.CreateRequest("/health")
+                .GetAsync();
+
+            var testLoggerprovider = server.Services.GetRequiredService<ILoggerProvider>() as TestLoggerProvider;
```
Comment (sungam3r) [inline test/HealthChecks.Npgsql.Tests/Functional/NpgsqlHealthCheckTests.cs:187]:
```suggestion
            var testLoggerProvider = (TestLoggerProvider)server.Services.GetRequiredService<ILoggerProvider>();

```

File: test/HealthChecks.Npgsql.Tests/Functional/NpgsqlHealthCheckTests.cs [hunk 3]
```diff
@@ -152,5 +154,42 @@ public async Task be_unhealthy_if_npgsql_is_not_available_registered()
...
+
+            using var server = new TestServer(webHostBuilder);
+
+            var response = await server.CreateRequest("/health")
+                .GetAsync();
+
+            var testLoggerProvider = (TestLoggerProvider)server.Services.GetRequiredService<ILoggerProvider>();
+
+            testLoggerprovider.Should().NotBeNull();
+            var logger = testLoggerprovider?.GetLogger("Microsoft.Extensions.Diagnostics.HealthChecks.DefaultHealthCheckService");
```
Comment (sungam3r) [inline test/HealthChecks.Npgsql.Tests/Functional/NpgsqlHealthCheckTests.cs:190]:
```suggestion
            testLoggerProvider.Should().NotBeNull();
            var logger = testLoggerProvider.GetLogger("Microsoft.Extensions.Diagnostics.HealthChecks.DefaultHealthCheckService");

```

File: src/HealthChecks.NpgSql/NpgSqlHealthCheck.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             }
             catch (Exception ex)
             {
-                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+                return new HealthCheckResult(context.Registration.FailureStatus, description: ex.Message, exception: ex);
```
Comment (sungam3r) [inline src/HealthChecks.NpgSql/NpgSqlHealthCheck.cs:40]: I see such usage only in `ArrangoDbHealthCheck` but I'm fine to pass exception message into description arg.

File: test/HealthChecks.Npgsql.Tests/Functional/TestLogger.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+using Microsoft.Extensions.Logging;
+
+namespace HealthChecks.Npgsql.Tests.Functional;
+
+public sealed class TestLogger : ILogger
```
Comment (sungam3r) [inline test/HealthChecks.Npgsql.Tests/Functional/TestLogger.cs:5]: I suggest to move `TestLogger` and `TestLoggerProvider` into `test` folder and and add `Compile Include="..."` in `Directory.Build.props` for each test project.

Comment (sungam3r) [reply test/HealthChecks.Npgsql.Tests/Functional/TestLogger.cs:5]: See `<ItemGroup Condition="$(MSBuildProjectName.EndsWith('.Tests')) OR $(MSBuildProjectName) == 'UnitTests'">` in `Directory.Build.props`.

File: src/HealthChecks.NpgSql/NpgSqlHealthCheck.cs (hunk 4 above)
Comment (pitming) [reply src/HealthChecks.NpgSql/NpgSqlHealthCheck.cs:40]: from our usage with a postgresql db, we get a null message in our log if healthcheck is unable to reach or connect to database. It ease the debug to know why it didn't succeed

File: test/HealthChecks.Npgsql.Tests/Functional/TestLogger.cs (hunk 5 above)
Comment (pitming) [reply test/HealthChecks.Npgsql.Tests/Functional/TestLogger.cs]: done

File: Directory.Build.props [hunk 6]
```diff
@@ -55,6 +55,8 @@
     <ProjectReference Include="..\..\src\HealthChecks.UI.Client\HealthChecks.UI.Client.csproj" />
 
     <Compile Include="../ApiApprovalTests.cs" Link="ApiApprovalTests.cs" />
+    <Compile Include ="../TestLogger.cs" Link="TestLogger.cs"/>
+    <Compile Include ="../TestLoggerProvider.cs" Link="TestLoggerProvider.cs"/>
```
Comment (sungam3r) [inline Directory.Build.props:59]:
```suggestion
    <Compile Include="../TestLogger.cs" Link="TestLogger.cs"/>
    <Compile Include="../TestLoggerProvider.cs" Link="TestLoggerProvider.cs"/>

```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**